provider/cluster/service_transfer_hostnames.go

```diff
@@ -0,0 +1,286 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"github.com/ovrclk/akash/provider/event"
+	"github.com/ovrclk/akash/pubsub"
+	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"time"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+)
+
+type transferHostnamesRequest struct {
+	hostnames []string
+	destination dtypes.DeploymentID
+	errCh chan <- error
+}
+
+type transferHostnamesManager struct {
+	hostnames []string
+	ownerAddr sdktypes.Address
+	destinationID dtypes.DeploymentID
+	toPurge map[mtypes.LeaseID]*deploymentManager
+	destination []*deploymentManager
+
+	cancel context.CancelFunc
+
+	log log.Logger
+
+	client Client
+}
+
+var errDestintationDeploymentDoesNotExist = errors.New("destination deployment does not exist")
+var errTransferRunning = errors.New("hostname transfer already running")
+
+func (s *service) transferHostnames(req transferHostnamesRequest) error {
+	ownerAddr, err := req.destination.GetOwnerAddress()
```

Comment:
do this in the client thread [here](https://github.com/ovrclk/akash/pull/1299/files#diff-a55c682314e02a83146fa4a7a6c67dc87dda87d97559f5edbfdb3cae4ad14efeR219)